### PR TITLE
[Plugin] Encryptcord

### PR DESCRIPTION
**Pull request by Inbestigator**
(Resubmission after requested fixes)

Encryptcord allows you to have an end-to-end encrypted conversation with your friends.
## Simple interface
Groups are joined/started by clicking the lock icon in your chatbar.
image
image

The system uses a 4096 bit RSA keypair to ensure that your messages are safe. The keys are regenerated after every restart.

## Choose who you want to talk to
When a user joins the group, you can click `I don't want to talk to you!` to stop sending/receiving messages to/from them.
image

## Toggle between message modes
Once in a group you can click the lock icon to toggle between sending encrypted and unencrypted messages
image
image

### CODE REVIEW

File: src/plugins/encryptcord/index.tsx [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+// Send Temporary Message
+async function sendTempMessage(recipientId: string, attachment: string, content: string, dm: boolean = true) {
+    if (recipientId == UserStore.getCurrentUser().id) return;
+
+    const dmChannelId = dm ? await RestAPI.post({
+        url: `/users/@me/channels`,
+        body: {
+            recipient_id: recipientId,
+        },
+    }).then((response) => response.body.id) : recipientId;
```
Comment (dolfies) [inline src/plugins/encryptcord/index.tsx:286]: Do not manually call the create private channel endpoint.

Comment (Vendicated) [reply src/plugins/encryptcord/index.tsx:286]: account ban any%

Comment (Inbestigator) [reply src/plugins/encryptcord/index.tsx:286]: > Do not manually call the create private channel endpoint.

What should I use instead?

Comment (dolfies) [reply src/plugins/encryptcord/index.tsx:286]:
```js
await findByProps('openPrivateChannel').getOrEnsurePrivateChannel('userId')
```

**Comment by Oi-jojo** [comment]
how do you even encrypt the messages?

**Comment by Inbestigator** [comment]
> how do you even encrypt the messages?

```ts
export const encryptData = async (pemPublicKey, data) => {
    const publicKey = await importPemPublicKey(pemPublicKey);

    const chunkSize = 446;

    const encryptedChunks: any[] = [];
    const encoder = new TextEncoder();

    for (let i = 0; i < data.length; i += chunkSize) {
        const chunk = await data.substring(i, i + chunkSize);
        const encryptedChunk = await crypto.subtle.encrypt(
            {
                name: "RSA-OAEP",
            },
            publicKey,
            encoder.encode(chunk)
        );
        encryptedChunks.push(arrayBufferToBase64(encryptedChunk));
    }

    return encryptedChunks;
};```

**Comment by Vendicated** [comment]
i don't think this is a good idea to have... too niche, and if people want encryption, they should instead use something like matrix, xmpp, signal, etc. Having it on discord just makes little sence

you may still post it in the third party plugins channel if you'd like

and thank you for your work regardless!

**Comment by thecatontheceiling** [comment]
I fucking hate matrix and I'm addicted to discord and I also have to discuss government secrets - thank you for the plugin!

**Comment by programminglaboratorys** [comment]
> I fucking hate matrix and I'm addicted to discord and I also have to discuss government secrets - thank you for the plugin!

<:husk:1026532993923293184>

**Comment by TazeXL** [comment]
i really like the idea but genuinely have no idea how to download this

**Pull request closed without merging**